### DISCUSSION TITLE
Is lottie using a private API, _overrideSizeThatFits?

### DISCUSSION THREAD

**Discussion by josh-arnold-1** [airbnb/lottie-ios#2476] [General]
## Which Version of Lottie are you using?

Lottie 4.5.0

## Question

I saw lottie is using `_overrideSizeThatFits` for the SwiftUI view, as the `sizeThatFits` is only available for iOS 16+.
([link to lottie code](https://github.com/airbnb/lottie-ios/blob/ab4c5d98c1c0b63aa10be6c68f995076ea1ff4cc/Sources/Private/EmbeddedLibraries/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringViewRepresentable.swift#L47C8-L47C29))

- Is this considered a private API by Apple? 
- Has anyone ran into any issues using this? 
- Would Apple allow this usage when submitting to the app store?
- Could this code break on older iOS versions?

Thanks!

**Comment by calda** [comment]
We've been using this for some time and haven't heard reports of any issues.

**Reply by josh-arnold-1** [reply]
Thanks for the response! So I assume apps have been approved by app store review using this?

**Reply by calda** [reply]
Yes, hundreds (maybe thousands) of apps use Lottie in the App Store, including Airbnb.
